mathesar_ui/src/pages/database-connection/DatabaseConnectionsList.svelte

```diff
@@ -0,0 +1,120 @@
+<script lang="ts">
+  import { databases } from '@mathesar/stores/databases';
+  import AppendBreadcrumb from '@mathesar/components/breadcrumb/AppendBreadcrumb.svelte';
+  import { iconAddNew, iconDatabase } from '@mathesar/icons';
+  import {
+    DATABASE_CONNECTION_ADD_URL,
+    DATABASE_CONNECTION_LIST_URL,
+  } from '@mathesar/routes/urls';
+  import type { Database } from '@mathesar/AppTypes';
+  import { States } from '@mathesar/api/utils/requestUtils';
+  import ErrorBox from '@mathesar/components/message-boxes/ErrorBox.svelte';
+  import EntityContainerWithFilterBar from '@mathesar/components/EntityContainerWithFilterBar.svelte';
+  import { AnchorButton, Icon } from '@mathesar/component-library';
+  import { labeledCount } from '@mathesar/utils/languageUtils';
+  import { isSuccessfullyConnectedDatabase } from '@mathesar/utils/preloadData';
+  import DatabaseConnectionSkeleton from './DatabaseConnectionSkeleton.svelte';
+  import { makeSimplePageTitle } from '../pageTitleUtils';
+  import DatabaseConnectionItem from './DatabaseConnectionItem.svelte';
+
+  let filterQuery = '';
+
+  $: isPreloaded = $databases.preload;
+  $: allDatabases = $databases.data;
+  $: databasesLoadStatus = $databases.state;
+  $: databasesLoadError = $databases.error;
+
+  function filterDatabase(_databases: Database[], query: string) {
+    function isMatch(database: Database, q: string) {
+      if (!isSuccessfullyConnectedDatabase(database)) {
+        return database.name.toLowerCase().includes(q);
+      }
+      return (
+        database.name.toLowerCase().includes(q) ||
+        database.db_name.toLowerCase().includes(q)
+      );
+    }
```

Comment:
This inner function does not make use of any parent arguments. It would be better to move out of the parent function.